File: 백준/Bronze/2748. 피보나치 수 2/피보나치 수 2.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const solution = (input) => {
  let answer = [BigInt(0), BigInt(1)];

  for (let i = 2; i <= input; i++) {
    answer.push(BigInt(answer[i - 2]) + BigInt(answer[i - 1]));
  }

  return answer[input].toString();
};

let input = [];
let list = [];
rl.on("line", function (line) {
  input = line.toString();
}).on("close", function () {
  const answer = solution(input);
  console.log(answer);
  process.exit();
});

File: 백준/Bronze/2748. 피보나치 수 2/피보나치 수 2.py
# 피보나치 수 2
import sys


class Solution:
    def solve(self, n: int) -> int:
        dp = [0,1]
        for n in range(2, n+1):
            dp.append(dp[n-1] + dp[n-2])

        return dp[n]


def main():
    n = int(sys.stdin.readline().strip())
    sol = Solution()
    answer = sol.solve(n)
    print(answer)


if __name__ == '__main__':
    main()
